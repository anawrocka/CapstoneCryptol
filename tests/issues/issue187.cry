// Helper synonyms:
// type Char        = [8]
module Enigma where

enigmaLoop : {n}(fin n) => [n]
enigmaLoop = undefined

// Encryption/Decryption
enigma : {n, m} (fin n, fin m) => ([n], [m]) -> [m]
enigma (m, pt) = tail [ True | _ <- rcs ]
  where rcs = [ (m, True)] # 
              [ (enigmaLoop, True)
              | _     <- pt
              | (_,_) <- rcs
              ]
