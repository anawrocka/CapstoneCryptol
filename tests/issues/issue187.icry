:l issue187.cry
