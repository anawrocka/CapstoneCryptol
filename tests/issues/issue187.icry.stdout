Loading module Cryptol
Loading module Cryptol
Loading module Enigma
